akka-persistence/src/main/scala/akka/persistence/snapshot/local/LocalSnapshotStore.scala

```diff
@@ -58,13 +60,18 @@ private[persistence] class LocalSnapshotStore extends SnapshotStore with ActorLo
     saving -= metadata
   }
 
-  def delete(metadata: SnapshotMetadata): Unit = {
+  def delete(metadata: SnapshotMetadata): Future[Unit] = {
     saving -= metadata
-    snapshotFile(metadata).delete()
+    Future {
+      snapshotFile(metadata).delete()
+      ()
+    }(ExecutionContexts.sameThreadExecutionContext)
```

Comment:
Actually no, the `Future {}` is nicer because if an exception is thrown it's captured in the exception nicely